packages/web/src/pages/sign-up-page/SignUpRootPage.tsx

```diff
@@ -24,38 +31,82 @@ type SignUpRootState =
   | FinishProfileState
   | SelectGenreState
   | SelectArtistsState
-  | AppCtaState
+// | AppCtaState
+
+// TODO: type this correctly
+const determineAllowedRoute = (signUpData: any, routeAttempt: SignUpStep) => {
```

Comment:
this is where the "keep the train on the tracks" magic happens, 
pretty straightforward approach just checking for each data point in order to decide if the requested route is allowed